.github/workflows/build-and-test.yml

```diff
@@ -94,10 +94,16 @@ jobs:
           git clone https://github.com/precice/precice.git precice-core
           mkdir -p precice
           cp precice-core/src/precice/SolverInterface.hpp precice/SolverInterface.hpp
+          cd precice-core
+          mkdir build && cd build
+          cmake .. -DPRECICE_MPICommunication=OFF -DPRECICE_PETScMapping=OFF -DPRECICE_PythonActions=OFF -DBUILD_TESTING=OFF
```

Comment:
Do you really need this? The purpose of the mocking was that we don't have to install preCICE. The line above where we copy `precice-core/src/precice/SolverInterface.hpp` should be sufficient. Can you try removing these 3 lines again? 

But if it is needed for pkg-config to work properly I guess it is just as it is. As long as the mock implementation from `python-bindings/test/SolverInterface.cpp` is used properly it is fine.

Reply:
Actually, pkg-config should not be used here. You are running the tests and the tests don't use pkg-config for linking to preCICE, but the hard-coded path with the mock-version of the interface.

Reply:
For the mock tests we need to run cmake because the `SolverInterface.hpp` that we copy over, has an included header `Version.h`. This header file is a generated header file, so we need to run cmake to get the file. After the file is generated, for the mock tests we use the cflags from pkg-config to link to this generated header. So we actually do use pkg-config even for the mock testing, just not the library flags.

Reply:
Ah ok. So I guess this is what this whole PR was about. Makes sense :+1: